### PR TITLE
Create shop.js

### PR DESCRIPTION
**Pull request by Lord-Haji**
Dynamic Shop system . Adding Items to shop is much more easy now.

### CODE REVIEW

**Comment by Lord-Haji** [comment]
@CreaturePhil @TheFenderStory Look through this plz...

File: chat-plugins/shop.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+function logMoney(message) {
+	if (!message) return;
+	let file = path.join(__dirname, '../logs/money.txt');
+	let date = "[" + new Date().toUTCString() + "] ";
+	let msg = message + "\n";
+	fs.appendFile(file, date + msg);
+}
+
+function writShop() {
```
Comment (TheFenderStory) [inline chat-plugins/shop.js]: why not writeShop?

File: chat-plugins/shop.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
+/*
+* Credits - Lord Haji, HoeenCoder
+*/
+'use strict';
+
+const fs = require('fs');
+let writeJSON = true;
+const path = require('path');
+let Shop = {};
```
Comment (TheFenderStory) [inline chat-plugins/shop.js]: Not a huge fan of having to recreate this object everytime we hotpatch.

File: chat-plugins/economy.js [hunk 3]
```diff
@@ -52,87 +39,13 @@ function isMoney(money) {
...
 	if (!message) return;
 	let file = path.join(__dirname, '../logs/money.txt');
 	let date = "[" + new Date().toUTCString() + "] ";
 	let msg = message + "\n";
 	fs.appendFile(file, date + msg);
-}
-
-/**
- * Displays the shop
- *
```
Comment (TheFenderStory) [inline chat-plugins/economy.js]: You need to follow the comment scheme used throughout the rest of the project.

File: chat-plugins/shop.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
+/*
+* Credits - Lord Haji, HoeenCoder
+*/
```
Comment (TheFenderStory) [inline chat-plugins/shop.js]: You need to use the comment style found throughout the rest of the project.

**Comment by HoeenCoder** [comment]
Seems like @Lord-Haji is using SGs alternate ["eshop"](https://github.com/HoeenCoder/SpacialGaze/blob/master/chat-plugins/eShop.js) code in this PR. Im not against merging this in i just wish someone would of told me this is happening.

**Comment by HoeenCoder** [comment]
The reason SG uses a global.currencyName setup is so it name of the currency can be changed by any servers that use a copy of our repo.

The reason the shop files are seperate is because our main shop setup is private code so it cant go in economy.js. The eshop could be merged into economy.js here.

File: chat-plugins/economy.js [hunk 5]
```diff
@@ -52,87 +39,13 @@ function isMoney(money) {
  *
  * @param {String} message
  */
-function logMoney(message) {
+global.logMoney = function (message) {
```
Comment (Lord-Haji) [inline chat-plugins/economy.js]: ohh kk

File: chat-plugins/shop.js (hunk 4 above)
Comment (Lord-Haji) [inline chat-plugins/shop.js]: Well if u see this
Same Comment style is used here too
https://github.com/CreaturePhil/Showdown-Boilerplate/blob/master/chat-plugins/seticon.js

File: chat-plugins/shop.js (hunk 2 above)
Comment (Lord-Haji) [inline chat-plugins/shop.js]: wut else can i do here ?

Comment (TheFenderStory) [inline chat-plugins/shop.js]: Perhaps these objects can be constructed on startup in the main file?

**Comment by Lord-Haji** [comment]
OK ill do all the the stuff in same file ...

**Pull request closed without merging**